pkg/controller/usercleanup/user_cleanup_controller_test.go

```diff
@@ -0,0 +1,132 @@
+package usercleanup
+
+import (
+	"context"
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/host-operator/pkg/configuration"
+	"github.com/codeready-toolchain/host-operator/pkg/metrics"
+	test2 "github.com/codeready-toolchain/host-operator/test"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"testing"
+	"time"
+
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func TestUserCleanup(t *testing.T) {
+	t.Run("test that user cleanup doesn't delete an active UserSignup", func(t *testing.T) {
+		userSignup := &v1alpha1.UserSignup{
+			ObjectMeta: test2.NewUserSignupObjectMeta("", "abigail.thompson@redhat.com"),
+			Spec: v1alpha1.UserSignupSpec{
+				UserID:      "User-98887",
+				Username:    "abigail.thompson@redhat.com",
+				Deactivated: false,
+			},
+			Status: v1alpha1.UserSignupStatus{
+				Conditions: []v1alpha1.Condition{
+					{
+						Type:   v1alpha1.UserSignupComplete,
+						Status: v1.ConditionTrue,
+					},
+					{
+						Type:   v1alpha1.UserSignupApproved,
+						Status: v1.ConditionTrue,
+						Reason: "ApprovedAutomatically",
+					},
+				},
+				CompliantUsername: "abigail-thompson",
+			},
+		}
+		userSignup.Labels["toolchain.dev.openshift.com/approved"] = "true"
+		userSignup.Labels[v1alpha1.UserSignupStateLabelKey] = "approved"
+
+		r, req, _ := prepareReconcile(t, userSignup.Name, userSignup)
+
+		_, err := r.Reconcile(req)
+		require.NoError(t, err)
+
+		// Confirm the UserSignup still exists
+		key := test.NamespacedName(test.HostOperatorNs, userSignup.Name)
+		require.NoError(t, r.client.Get(context.Background(), key, userSignup))
+		require.NotNil(t, userSignup)
+	})
+
+	t.Run("test that user cleanup doesn't delete a recently deactivated UserSignup", func(t *testing.T) {
```

Comment:
Please also add tests for:
- old deactivated signup deleted
- old unverified signup deleted
- old (it has to be old) verified but unapproved is not deleted

Reply:
Done